src/app/components/named-entities-list/named-entities-list.component.ts

```diff
@@ -0,0 +1,53 @@
+import { Component, EventEmitter, Input, OnChanges, OnInit, Output } from '@angular/core';
+import { NamedEntitiesList } from '../../models/evt-models';
+import { EVTBtnClickEvent } from '../../ui-components/button/button.component';
+
+@Component({
+  selector: 'evt-named-entities-list',
+  templateUrl: './named-entities-list.component.html',
+  styleUrls: ['./named-entities-list.component.scss'],
+})
+export class NamedEntitiesListComponent implements OnInit, OnChanges {
+  @Input() list: NamedEntitiesList;
+  @Output() searchEntities: EventEmitter<string> = new EventEmitter();
+  public selectedKey: string;
+  public navigationKeys: string[] = [];
+
+  public searchOpened = false;
+  public querySearch = '';
+  public querySearchSubmitted = '';
+  public caseSensitiveSearch = false;
+
+  ngOnInit() {
+    this.initKeys();
+  }
+
+  ngOnChanges() {
+    this.initKeys();
+  }
+
+  selectKey(key: string) {
+    this.selectedKey = key;
+  }
+
+  toggleSearch() {
+    this.searchOpened = !this.searchOpened;
+    this.querySearch = '';
+    this.querySearchSubmitted = '';
+    this.searchEntities.emit(this.querySearch);
+  }
+
+  toggleCaseSensitiveSearch(event: EVTBtnClickEvent) {
+    this.caseSensitiveSearch = event.active;
+  }
+
+  private initKeys() {
+    if (this.list) {
+      this.navigationKeys = this.list.content
+        .map(el => el.sortKey.substr(0, 1).toLowerCase())
+        .filter((item, i, ar) => ar.indexOf(item) === i)
+        .sort();
+      this.selectKey(this.navigationKeys[0]);
```

Comment:
Can `this.list.content` have length `0` or be `undefined`?

Reply:
`this.list.content` is always initialized as an empty array, thus it can have length `0`.
I'll handle the case.